lib/Parser/Parse.cpp

```diff
@@ -8443,6 +8474,19 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             // Parse the operand, make a new node, and look for more
             pnodeT = ParseExpr<buildAST>(opl, NULL, fAllowIn, FALSE, pNameHint, &hintLength, &hintOffset, nullptr);
 
+            // Detect nested function escapes of the pattern "o.f = function(){...}" or "o[s] = function(){...}".
+            // Doing so in the parser allows us to disable stack-nested-functions in common cases where an escape
+            // is not detected at byte code gen time because of deferred parsing.
+            if (m_currentNodeFunc && pnodeT && pnodeT->nop == knopFncDecl && fIsDotOrIndex && nop == knopAsg)
```

Comment:
Is this m_currentNodeFunc be null when we are at deferred parsing? If yes then how are we making sure to call SetNestedFuncEscapes?